repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,261 @@
+            } else {
+              logger.warn("Cannot write to DOT file: " + dotPath)
+            }
+          } catch {
+            case ex: IOException =>
+              logger.warn("Cannot write to DOT file: " + dotPath, ex)
+          }
+        }
+      case _ =>
+    }
+    new DispatchHandler(handlers.toList:_*)
+  }
+
+  private def getPath(path: String, configRoot: String): String = {
+    val file: File = new File(path)
+    if (file.isAbsolute) {
+      file.getAbsolutePath
+    } else {
+      new File(configRoot, path).getAbsolutePath
+    }
+  }
+
+  def readResource(resourceStream: InputStream): Option[String] = {
+    Try(Some(Source.fromInputStream(resourceStream).getLines().mkString("\n"))).getOrElse(None)
+  }
+
+  private def rbacToWadl(rbac: Option[String]): Option[String] = {
+
+    case class Resource(path: String, methods: Set[String], roles: Set[String])
+
+    def parseLine(line: String): Option[Resource] = {
+      val values = line.split("\\s+")
+      if (values.length == 3) {
+        Some(new Resource(
+          values(0),
+          Try(values(1).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty),
+          Try(values(2).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty)
+        ))
+      } else {
+        logger.warn(s"Malformed RBAC Resource: $line")
+        None
+      }
+    }
+
+    val parsed = rbac match {
+      case Some(l) =>
+        Some(l.replaceAll("[\r?\n?]", "\n").split('\n').toList.flatMap(parseLine))
+      case _ =>
+        None
+    }
+
+    parsed match {
+      case Some(_) =>
+        val header = s"""<application xmlns:rax="http://docs.rackspace.com/api"
+                        |         xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+                        |         xmlns="http://wadl.dev.java.net/2009/02"
+                        |    >
+                        |  <resources base="http://localhost">""".stripMargin
+        val resources = parsed.getOrElse(List.empty[Resource]).map { r =>
+          def toMethods(resource: Resource, uuid: UUID) = {
+            val roles = resource.roles.mkString(" ")
+            val raxRoles = roles match {
+              case a if roles.equalsIgnoreCase("ANY") || roles.equalsIgnoreCase("ALL") => ""
+              case _ =>
+                s"""rax:roles="$roles""""
+            }
+            resource.methods.map {
+              _ match {
+                case a if a.equalsIgnoreCase("ANY") || a.equalsIgnoreCase("ALL") =>
+                  s"""      <method name="GET"    id="_$uuid-GET"    $raxRoles/>
+                     |      <method name="PUT"    id="_$uuid-PUT"    $raxRoles/>
+                     |      <method name="POST"   id="_$uuid-POST"   $raxRoles/>
+                     |      <method name="DELETE" id="_$uuid-DELETE" $raxRoles/>""".stripMargin
+                case m =>
+                  s"""      <method name="$m"   id="_$uuid-$m"    $raxRoles/>"""
```

Comment:
API Checker is case sensitive on both methods and roles. This was updated to always put uppercase HTTP methods in the generated WADL.